cumulus_etl/etl/tasks/basic_tasks.py

```diff
@@ -0,0 +1,129 @@
+"""Standard FHIR resource tasks"""
+
+import copy
+import logging
+import os
+
+from cumulus_etl import common, fhir, store
+from cumulus_etl.etl import tasks
+
+
+class ConditionTask(tasks.EtlTask):
+    name = "condition"
+    resource = "Condition"
+    tags = {"cpu"}
+
+
+class DocumentReferenceTask(tasks.EtlTask):
+    name = "documentreference"
+    resource = "DocumentReference"
+    tags = {"cpu"}
+
+
+class EncounterTask(tasks.EtlTask):
+    name = "encounter"
+    resource = "Encounter"
+    tags = {"cpu"}
+
+
+class MedicationRequestTask(tasks.EtlTask):
+    name = "medicationrequest"
+    resource = "MedicationRequest"
+    tags = {"cpu"}
+
+    # We may write to a second Medication table as we go.
+    # MedicationRequest can have inline medications via CodeableConcepts, or external Medication references.
+    # If external, we'll download them and stuff them in this output table.
+    # We do all this special business logic because Medication is a special, "reference" resource,
+    # and many EHRs don't let you simply bulk export them.
+
+    outputs = [
+        tasks.OutputTable(name="medication", schema="Medication"),
+        tasks.OutputTable(),
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Keep a cache of medication IDs that we've already downloaded.
+        # If this ends up growing too large in practice, we can wipe it during a call to table_batch_cleanup().
+        # But let's try initially with keeping it around for the whole task.
+        self.medication_ids = set()
+
+    def scrub_medication(self, medication: dict) -> bool:
```

Comment:
This method is maybe something to look at with side-eye. I'm doing some manual de-identification (rather than via the MS tool).

I figured this made sense for technical reasons (MS tool only operates on a pile of ndjson on disk, and normally only runs once at the beginning of ETL) and it-isn't-patient-data reasons (these resources have no connection to patients at all).

But still... I'm open to feedback. I *could* do it via MS tool by writing all these medications to disk and then running it on them, then reading that back in.

Here's the spec for this resource, FYI: https://www.hl7.org/fhir/R4/medication.html

Reply:
For posterity: I checked with Jamie and Andy, they think this is OK.

But also, in further chats with you, we may want to allow ingesting Medications like normal resources anyway -- for (A) future EHRs that do let you bulk export them and (B) cases like Synthea where you have pre-generated resources.

In those cases, we may want to switch to an MS config for both flows and write out all Medications to a local ndjson and re-ingest it.

But for now, let's go with this and manually deal with Synthea tables.